### ISSUE TITLE
Switch from `fail` to `throwCrucibleSetup` for error handling in crucible-jvm commands

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/saw-script#928]
The crucible-jvm verification commands use `fail` all over the place to signal failure. We should be using `throwCrucibleSetup` instead, like the LLVM commands use. One particular benefit is its uniform handling of location information.

**Comment by chameco** [comment]
:+1: 

Note that currently the location information attached to the exception is not used by the handler in `SAWScript.Value`, since the exception message usually makes it redundant. As we move away from `fail`, it might be nice to consider attaching more context to exceptions and then printing this context in a consistent way in the handler, vs. throwing large strings. This would let us more easily e.g. emit JSON errors instead of plaintext, add CLI options to include/exclude specific information like locations, etc.

**Comment by brianhuffman** [comment]
Yes, I definitely like the idea of more structured exception types for user errors. Another benefit of this is that the rendering of the error strings can be done all in one place in the code, which makes it easier to ensure that the error message formatting and content is consistent.

**Comment by brianhuffman** [comment]
PR #948 removed a large number of uses of `fail` in the SAW/JVM code: All errors that can be thrown by `JVMSetup` commands now use the custom `JVMSetupError` exception type.

However, there are still many uses of `fail` in modules `SAWScript.Crucible.JVM.ResolveSetupValue`, `SAWScript.Crucible.JVM.Override`, `SAWScript.Crucible.JVM.Builtins`, and `SAWScript.JVM.BuiltinsJVM`. We can close this ticket once we get replace all of those with `throwM` or `panic`, as appropriate.

**Comment by brianhuffman** [comment]
All the uses of `fail` in the `JVMPointsToArray` case of `learnPointsTo` should be changed to `panic`, as all the conditions checked here are datatype invariants that are already checked in `jvm_array_is` when the value is created: https://github.com/GaloisInc/saw-script/blob/c4ab82b939f62a83f9de91b99822e05355e53803/src/SAWScript/Crucible/JVM/Override.hs#L695-L704
